### ISSUE TITLE
rr gdb session shows the current environment, not the environment of the recording

### ISSUE THREAD

**Issue by rrnewton** [rr-debugger/rr#2764]
I was thinking that `gdb`'s "show environment" would be a great way to see the HOSTNAME that a recording came from.  No such luck, it just shows me the hostname that I'm replaying on.

Isn't that a bug?  I.e. the intent of the time-travel debugging session is to accurately simulate being in the time and place the software was recorded...

To reproduce:

```
$ rr record ls
...
$ HOSTNAME=foobar rr replay
...
$ show environment
...
HOSTNAME=foobar
```

Expected: the real hostname above instead of "foobar".

**Comment by khuey** [comment]
Yes, this is a bug, but unfortunately it's in gdb. As far as I know there is no way to tell gdb what the environment is via the gdbserver protocol. We would need gdb to support that in the protocol and then we could add whatever is needed on our end to make it work.

As a workaround you can examine the `environ` global variable (in a C program anyways) directly. It's an array of char*s that point to `HOSTNAME=foobar` type strings, and the very last entry is null.
